src/google/protobuf/compiler/command_line_interface.cc

```diff
@@ -188,8 +188,25 @@ bool TryCreateParentDirectory(const string& prefix, const string& filename) {
 // Get the absolute path of this protoc binary.
 bool GetProtocAbsolutePath(string* path) {
 #ifdef _WIN32
-  char buffer[MAX_PATH];
-  int len = GetModuleFileName(NULL, buffer, MAX_PATH);
+  wchar_t wide_buffer[MAX_PATH];
+  // Note: the returned length does not include terminating null character.
+  int len = GetModuleFileNameW(NULL, wide_buffer, MAX_PATH);
+  if (len == GOOGLE_ARRAYSIZE(wide_buffer) &&
+      GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+    // The path is too long.
+    return false;
+  }
+  char buffer[MAX_PATH + 1];
+  for (size_t i = 0; i < len; ++i) {
+    wchar_t wide_char = wide_buffer[i];
+    if (wide_char > 0x7F) {
+      // non-ASCII character is not supported yet.
+      // TODO(yukawa): Support paths that contain non-ASCII characters.
```

Comment:
I suspect GetModuleFileNameA() will actually work for non-ASCII characters. Could you help check?